WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
```

Comment:
I think a Service class should only operate with its own context. Passing objects from another context shouldn't happen (although it can)

To be safer, I think the Service classes should only take object IDs as arguments, and fetch objects on their own contexts, but that's more of an architecture discussion, and not in the scope of this PR. For this, I'd rather follow the patterns of other Service classes